frontend/src/components/UserFeedPage/CreatPost.js

```diff
@@ -91,31 +92,36 @@ function CreatPost() {
         } else {
           const connections = (await getDoc(doc(collection(db, "connection"), currentId))).data().connections
           connections.forEach(id => {
-            getDoc(doc(collection(db, 'Notifications'), id)).then((followed_doc) => {
-              const note = {
-                message: `${user_info.first_name} ${user_info.last_name} has created a new post, go check it out!`,
-                profilePicUrl: user_info.profile_picture,
-                id: generateKey(8),
-                post_id:post.id
-              }
-              if ((followed_doc.data() === undefined) || (followed_doc.data().notifications.length === 0)) {
-                setDoc(doc(collection(db, 'Notifications'), id), { notifications: [note] })
-              } else {
-                const notifications_array = followed_doc.data().notifications;
-                let condition = false
-                notifications_array.forEach((notif) => {
-                  if (!(notif.id === note.id)) {
-                    condition = true;
-                  }
-                })
-                if (condition) {
-                  notifications_array.push(note)
+            getDoc(doc(collection(db, "notification_settings"), id)).then((note_data) => {
+              if (note_data.data() !== undefined) {
```

Comment:
I see you made it more robust by checking if there's an undefined value. Good thinking.